openproblems/tasks/denoising/methods/knn_smoothing.py

```diff
@@ -16,7 +16,8 @@ def knn_smoothing(adata, test=False):
     import numpy as np
 
     adata.uns["method_code_version"] = check_version("knn_smooth")
+    X = np.float64(np.array(adata.obsm["train"].transpose()))

```

Comment:
You'll want `adata.obsm["train"].toarray()` to convert to numpy and `X.astype(np.float64)` to convert to float64.

Reply:
Changed to:
X = adata.obsm["train"].transpose().toarray()
X = X.astype(np.float64)